### PR TITLE
Match api 2

### PR DESCRIPTION
**Pull request by Shidesu**
As you told me @BenFradet, I opened a new PR + I modified the Program.cs file.

### CODE REVIEW

**Comment by BenFradet** [comment]
@Shidesu great, will try to have a look today

**Comment by Shidesu** [comment]
@BenFradet I've found an issue with the deserialization -> impossible to assign value to the season field of MatchReference, but can't find how to fix it.

File: RiotSharp/Interfaces/IRiotApi.cs [hunk 1]
```diff
@@ -208,75 +207,88 @@ public interface IRiotApi
         #endregion
 
         #region Match
+
+        List<long> GetMatchIdsByTournamentCode(Region region, string tournamentCode);
```
Comment (BenFradet) [inline RiotSharp/Interfaces/IRiotApi.cs]: could we add some doc there?

File: RiotSharp/Interfaces/IRiotApi.cs [hunk 2]
```diff
@@ -208,75 +207,88 @@ public interface IRiotApi
...
+
+        Task<List<long>> GetMatchIdsByTournamentCodeAsync(Region region, string tournamentCode);
```
Comment (BenFradet) [inline RiotSharp/Interfaces/IRiotApi.cs]: same

File: RiotSharp/MatchEndpoint/Match.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        public string GameType { get; set; }
+
+        [JsonProperty("teams")]
+        public List<TeamStats> Teams { get; set; }
+
+        [JsonProperty("participants")]
+        public List<Participant> Participants { get; set; } 
+
+        [JsonProperty("gameDuration")]
+        [JsonConverter(typeof(TimeSpanConverterFromMilliseconds))] // Correct?
```
Comment (BenFradet) [inline RiotSharp/MatchEndpoint/Match.cs]: correct, please rm the comment

File: RiotSharp/MatchEndpoint/MatchReference.cs [hunk 4]
```diff
@@ -23,14 +23,14 @@ public class MatchReference
...
-        [JsonProperty("matchId")]
-        public long MatchID { get; set; }
+        [JsonProperty("gameId")]
+        public long GameId { get; set; }
 
         /// <summary>
         /// The ID of the platform on which the game is being played
         /// </summary>
         [JsonProperty("platformId")]
-        public string PlatformID { get; set; }
```
Comment (BenFradet) [inline RiotSharp/MatchEndpoint/MatchReference.cs]: isn't the converter needed?

File: RiotSharp/MatchEndpoint/ParticipantTimeline.cs [hunk 5]
```diff
@@ -1,176 +1,40 @@
...
 namespace RiotSharp.MatchEndpoint
 {
-    /// <summary>
-    /// Class representing a participant's timeline (Match API).
-    /// </summary>
     public class ParticipantTimeline
     {
-        internal ParticipantTimeline() { }
+        [JsonProperty("lane")]
+        public string Lane { get; set; }
```
Comment (BenFradet) [inline RiotSharp/MatchEndpoint/ParticipantTimeline.cs]: can we have some doc in this file?

File: RiotSharp/MatchEndpoint/ParticipantTimeline_OLD.cs [hunk 6]
```diff
@@ -0,0 +1,176 @@
+﻿using Newtonsoft.Json;
```
Comment (BenFradet) [inline RiotSharp/MatchEndpoint/ParticipantTimeline_OLD.cs]: to rm

File: RiotSharp/MatchEndpoint/Player.cs [hunk 7]
```diff
@@ -10,6 +10,17 @@ public class Player
         internal Player() { }
 
         /// <summary>
+        /// Current platform ID.
+        /// </summary>
+        [JsonProperty("currentPlatformId")]
+        public string CurrentPlatformId { get; set; }
```
Comment (BenFradet) [inline RiotSharp/MatchEndpoint/Player.cs]: can't we leverage platform here?

File: RiotSharp/MatchEndpoint/Player.cs [hunk 8]
```diff
@@ -10,6 +10,17 @@ public class Player
...
+
+        /// <summary>
+        /// Platform ID.
+        /// </summary>
+        public string PlatformId { get; set; }
```
Comment (BenFradet) [inline RiotSharp/MatchEndpoint/Player.cs]: here too?

File: RiotSharp/MatchEndpoint/Player.cs [hunk 9]
```diff
@@ -22,6 +33,17 @@ public class Player
...
         /// <summary>
+        /// Current account ID.
+        /// </summary>
+        [JsonProperty("currentAccountId")]
+        public long CurrentAccountId { get; set; }
+
+        /// <summary>
+        /// Account ID.
+        /// </summary>
+        public long AccountId { get; set; }
```
Comment (BenFradet) [inline RiotSharp/MatchEndpoint/Player.cs]: the json property is missing

File: RiotSharp/MatchEndpoint/TeamBan.cs [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace RiotSharp.MatchEndpoint
+{
+    public class TeamBan
+    {
+        internal TeamBan() { }
+
+        [JsonProperty("pickTurn")]
```
Comment (BenFradet) [inline RiotSharp/MatchEndpoint/TeamBan.cs]: can we have some doc?

File: RiotSharp/MatchEndpoint/TeamStats.cs [hunk 11]
```diff
@@ -83,15 +94,15 @@ public class Team
...
 
         /// <summary>
-        /// Number of times the team killed vilemaw.
+        /// Number of times the team killed vilemaw (Twisted Treeline epic monster).
         /// </summary>
         [JsonProperty("vilemawKills")]
         public int VilemawKills { get; set; }
 
         /// <summary>
         /// Flag indicating whether or not the team won.
```
Comment (BenFradet) [inline RiotSharp/MatchEndpoint/TeamStats.cs]: I don't think the doc matches anymore

Comment (BenFradet) [inline RiotSharp/MatchEndpoint/TeamStats.cs]: I don't think the doc matches anymore

File: RiotSharp/MatchEndpoint/MatchReference.cs (hunk 4 above)
Comment (Shidesu) [reply RiotSharp/MatchEndpoint/MatchReference.cs]: This could explain the bogue in the Example program.

Comment (Shidesu) [reply RiotSharp/MatchEndpoint/MatchReference.cs]: I have no error about this so I don't think so 🤷‍♂️

File: RiotSharp/MatchEndpoint/ParticipantTimeline_OLD.cs (hunk 6 above)
Comment (Shidesu) [reply RiotSharp/MatchEndpoint/ParticipantTimeline_OLD.cs]: The file or the line ?

Comment (Shidesu) [reply RiotSharp/MatchEndpoint/ParticipantTimeline_OLD.cs]: The file, right ?

Comment (BenFradet) [reply RiotSharp/MatchEndpoint/ParticipantTimeline_OLD.cs]: yes :+1:

File: RiotSharp/MatchEndpoint/MatchReference.cs (hunk 4 above)
Comment (BenFradet) [reply RiotSharp/MatchEndpoint/MatchReference.cs]: mmmh

File: RiotSharp/MatchEndpoint/Player.cs (hunk 7 above)
Comment (Shidesu) [reply RiotSharp/MatchEndpoint/Player.cs]: I'm not sure ton understand the meaning.

Comment (BenFradet) [reply RiotSharp/MatchEndpoint/Player.cs]: I mean replace the string by https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/Misc/Platform.cs

File: RiotSharp/MatchEndpoint/MatchReference.cs (hunk 4 above)
Comment (Shidesu) [reply RiotSharp/MatchEndpoint/MatchReference.cs]: I added it in the dbae593 commit.

File: RiotSharp/MatchEndpoint/Player.cs (hunk 8 above)
Comment (Shidesu) [reply RiotSharp/MatchEndpoint/Player.cs]: Should I add the Json property too ?

Comment (BenFradet) [reply RiotSharp/MatchEndpoint/Player.cs]: yup :+1:

File: RiotSharp/MatchEndpoint/MatchReference.cs (hunk 4 above)
Comment (Shidesu) [reply RiotSharp/MatchEndpoint/MatchReference.cs]: I don't know why it's not updated here 🤷‍♂️

Comment (BenFradet) [reply RiotSharp/MatchEndpoint/MatchReference.cs]: yeah the review is a bit finicky when you change something just below where the commit is

File: RiotSharp/MatchEndpoint/TeamStats.cs (hunk 11 above)
Comment (Shidesu) [reply RiotSharp/MatchEndpoint/TeamStats.cs]: Why are you saying that ?

Comment (BenFradet) [reply RiotSharp/MatchEndpoint/TeamStats.cs]: because it's not a flag anymore since it's a string

Comment (Shidesu) [reply RiotSharp/MatchEndpoint/TeamStats.cs]: Oh the review was for the flag. It was not clear because the "centered" lines were the ones about Vilemaw.

Comment (BenFradet) [reply RiotSharp/MatchEndpoint/TeamStats.cs]: yes I try to comment about the line just above :+1:

**Comment by BenFradet** [comment]
could you have a look @JanOuborny ?

**Comment by JanOuborny** [comment]
@BenFradet I was waiting for the comments to resolve.

**Comment by BenFradet** [comment]
@Shidesu could you take care of https://github.com/BenFradet/RiotSharp/pull/473#discussion-diff-141465900R103 I think it's the last item on my review and rebase against develop to solve the conflicts?

**Comment by Shidesu** [comment]
I'm sorry, I was really busy, I'm gonna try to do it now 👍

**Comment by Shidesu** [comment]
I think it's okay now @BenFradet

**Comment by BenFradet** [comment]
@Shidesu Thanks!

@JanOuborny could you have a look?

**Comment by JanOuborny** [comment]
@BenFradet I will try to do it today.

File: RiotSharp/Interfaces/IRiotApi.cs [hunk 12]
```diff
@@ -209,74 +208,98 @@ public interface IRiotApi
...
         /// Get match information about a specific match synchronously.
         /// </summary>
         /// <param name="region">Region in which the match took place.</param>
         /// <param name="matchId">The match ID to be retrieved.</param>
-        /// <param name="includeTimeline">Whether or not to include timeline information.</param>
-        /// <returns>A match detail object containing information about the match.</returns>
-        MatchDetail GetMatch(Region region, long matchId, bool includeTimeline = false);
+        /// <param name="accountId">If provided, used to identify the participant to be unobfuscated.</param>
+        /// <returns>A match object containing information about the match.</returns>
+        Match GetMatch(Region region, long matchId, long? accountId = null);
```
Comment (JanOuborny) [inline RiotSharp/Interfaces/IRiotApi.cs]: As far as I know Riot removed the `accountId` parameter. Ref: https://developer.riotgames.com/api-methods/#match-v3/GET_getMatch

File: RiotSharp/Interfaces/IRiotApi.cs [hunk 13]
```diff
@@ -209,74 +208,98 @@ public interface IRiotApi
...
         /// Get match information about a specific match asynchronously.
         /// </summary>
         /// <param name="region">Region in which the match took place.</param>
         /// <param name="matchId">The match ID to be retrieved.</param>
-        /// <param name="includeTimeline">Whether or not to include timeline information.</param>
-        /// <returns>A match detail object containing information about the match.</returns>
-        Task<MatchDetail> GetMatchAsync(Region region, long matchId, bool includeTimeline = false);
+        /// <param name="accountId">If provided, used to identify the participant to be unobfuscated.</param>
+        /// <returns>A match object containing information about the match.</returns>
+        Task<Match> GetMatchAsync(Region region, long matchId, long? accountId = null);
```
Comment (JanOuborny) [inline RiotSharp/Interfaces/IRiotApi.cs]: same here

**Comment by JanOuborny** [comment]
Sorry for the late review. I started my math studies and ran out of time.

**Comment by Shidesu** [comment]
Alright, will check and correct this certainly tomorrow.

**Comment by Shidesu** [comment]
I'm back, sorry for being a bit late.

**Comment by Shidesu** [comment]
There was a problem with the first rebase don't know why. Should be okay now.

File: RiotSharp/RiotApi.cs [hunk 14]
```diff
@@ -334,123 +341,92 @@ public League GetMasterLeague(Region region, string queue)
...
-                    ? new List<string> {$"includeTimeline={includeTimeline.ToString().ToLower()}"}
-                    : null);
+            var json = await requester.CreateGetRequestAsync(MatchRootUrl +
+                                                             string.Format(MatchIdsByTournamentCodeUrl, tournamentCode),
+                region);
+
+            return await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<List<long>>(json));
+        }
+
+        public Match GetMatch(Region region, long matchId, long? accountId = null)
```
Comment (BenFradet) [inline RiotSharp/RiotApi.cs]: I think you forgot to remove the `accountId` here based on @JanOuborny 's comments

File: RiotSharp/RiotApi.cs [hunk 15]
```diff
@@ -334,123 +341,92 @@ public League GetMasterLeague(Region region, string queue)
...
+        {
+            var json = requester.CreateGetRequest(MatchRootUrl +
+                                                  string.Format(MatchByIdUrl, matchId), region,
+                accountId.HasValue ? new List<string> {string.Format("forAccountId={0}", accountId)} : null);
+            return JsonConvert.DeserializeObject<Match>(json);
+        }
+
+        public async Task<Match> GetMatchAsync(Region region, long matchId, long? accountId = null)
```
Comment (BenFradet) [inline RiotSharp/RiotApi.cs]: same

**Comment by JanOuborny** [comment]
When @BenFradet's change requests are done, it's good to go.

**Comment by Shidesu** [comment]
What the hell... I did it uh. Let's do it again.

**Comment by Shidesu** [comment]
I miss to delete something I'm sorry @BenFradet !

**Comment by BenFradet** [comment]
What do you mean?

**Comment by Shidesu** [comment]
About lines 364 to 377 in RiotApi.cs
These lines should be like this : `public Match GetMatch(Region region, long matchId)
        {
            var json = requester.CreateGetRequest(MatchRootUrl +
                                                  string.Format(MatchByIdUrl, matchId), region);
            return JsonConvert.DeserializeObject<Match>(json);
        }

        public async Task<Match> GetMatchAsync(Region region, long matchId)
        {
            var json = requester.CreateGetRequest(MatchRootUrl +
                                                  string.Format(MatchByIdUrl, matchId), region);
            return await Task.Factory.StartNew(() =>
                JsonConvert.DeserializeObject<Match>(json));
        }` and they are not.

**Comment by BenFradet** [comment]
then can you open another pr please?

**Comment by Shidesu** [comment]
Alright wait a minute.

**Pull request merged**